Time Master/script.js

```diff
@@ -0,0 +1,147 @@
+// 1 Time object basics
+// Date object to represent the current time
+const currentTime = new Date();
+
+// Extracting hours, minutes, and seconds from the Date object
+const hours = currentTime.getHours();
+const minutes = currentTime.getMinutes();
+const seconds = currentTime.getSeconds();
+
+// Logging the current time
+console.log(`Current Time: ${hours}:${minutes}:${seconds}`);
+
+// 2 Object Oriented Clock
+// Task: Design a Clock object with properties like hours, minutes, and seconds
+
+class Clock {
+  constructor() {
+    // Date object to get current time
+    const currentTime = new Date();
+
+    // Setting properties for hours, minutes, and seconds
+    this.hours = currentTime.getHours();
+    this.minutes = currentTime.getMinutes();
+    this.seconds = currentTime.getSeconds();
+  }
+
+  // Method to return time in "HH:MM:SS" format (24-hour format)
+  getFormattedTime() {
+    const formattedHours = this.hours.toString().padStart(2, "0");
+    const formattedMinutes = this.minutes.toString().padStart(2, "0");
+    const formattedSeconds = this.seconds.toString().padStart(2, "0");
+
+    return `${formattedHours}:${formattedMinutes}:${formattedSeconds}`;
+  }
+
+  // Method to return time in 12-hour format with AM/PM
+  get12HourTime() {
+    let period = "AM";
+    let hours12 = this.hours;
+
+    // Converting 24-hour time to 12-hour time
+    if (hours12 >= 12) {
+      period = "PM";
+      hours12 -= 12; // Subtracting 12 to convert to 12-hour format
+    }
+
+    // Handling the case for midnight (00:00), which should be 12 AM
+    if (hours12 === 0) {
+      hours12 = 12;
+    }
+
+    const formattedHours = hours12.toString().padStart(2, "0");
+    const formattedMinutes = this.minutes.toString().padStart(2, "0");
+    const formattedSeconds = this.seconds.toString().padStart(2, "0");
+

```

Comment:
For clean coding, avoiding repetitions is vital. I think it is better to use the getFormattedTime method inside this method, since it is doing the same task the above lines are doing.